src/main/java/kitchenpos/application/OrderService.java

```diff
@@ -1,108 +1,89 @@
-    private final OrderDao orderDao;
-    private final OrderLineItemDao orderLineItemDao;
-    private final OrderTableDao orderTableDao;
+    private final MenuService menuService;
+    private final OrderTableRepository orderTableRepository;
+    private final OrderRepository orderRepository;
 
     public OrderService(
-            final MenuDao menuDao,
-            final OrderDao orderDao,
-            final OrderLineItemDao orderLineItemDao,
-            final OrderTableDao orderTableDao
+            final MenuService menuService,
+            final OrderTableRepository orderTableRepository,
+            final OrderRepository orderRepository
     ) {
-        this.menuDao = menuDao;
-        this.orderDao = orderDao;
-        this.orderLineItemDao = orderLineItemDao;
-        this.orderTableDao = orderTableDao;
+        this.menuService = menuService;
+        this.orderTableRepository = orderTableRepository;
+        this.orderRepository = orderRepository;
     }
 
     @Transactional
-    public Order create(final Order order) {
-        final List<OrderLineItem> orderLineItems = order.getOrderLineItems();
+    public Order create(final OrderRequest orderRequest) {
+        final OrderTable orderTable = findOrderTableById(orderRequest.getOrderTableId());
 
-        if (CollectionUtils.isEmpty(orderLineItems)) {
-            throw new IllegalArgumentException();
-        }
-
-        final List<Long> menuIds = orderLineItems.stream()
-                .map(OrderLineItem::getMenuId)
-                .collect(Collectors.toList());
-
-        if (orderLineItems.size() != menuDao.countByIdIn(menuIds)) {
-            throw new IllegalArgumentException();
-        }
-
-        final OrderTable orderTable = orderTableDao.findById(order.getOrderTableId())
-                .orElseThrow(IllegalArgumentException::new);
-
-        if (orderTable.isEmpty()) {
-            throw new IllegalArgumentException();
-        }
-
-        order.setOrderTableId(orderTable.getId());
-        order.setOrderStatus(OrderStatus.COOKING.name());
-        order.setOrderedTime(LocalDateTime.now());
-
-        final Order savedOrder = orderDao.save(order);
-
-        final Long orderId = savedOrder.getId();
         final List<OrderLineItem> savedOrderLineItems = new ArrayList<>();
-        for (final OrderLineItem orderLineItem : orderLineItems) {
-            orderLineItem.setOrderId(orderId);
-            savedOrderLineItems.add(orderLineItemDao.save(orderLineItem));
+        for (final OrderLineItemRequest request : getOrderLineItemRequests(orderRequest)) {
+            Menu menu = menuService.findById(request.getMenuId());
+            OrderLineItem orderLineItem = new OrderLineItem(menu, request.getQuantity());
+            savedOrderLineItems.add(orderLineItem);
         }
-        savedOrder.setOrderLineItems(savedOrderLineItems);
 
-        return savedOrder;
+        return orderRepository.save(new Order(orderTable, OrderStatus.COOKING, savedOrderLineItems));
```

Comment:
```suggestion
        return orderRepository.save(new Order(orderTable, savedOrderLineItems));
```

Order 생성자를 쓰는 곳이 없으니 이렇게 바꿔도 되지 않을까요?
생성한 객체의 기본 상태를 표현해주려면 팩토리 메서드를 활용해서 표현하는 방법도 있겠네요

```java
Order.ofCoking(orderTable, savedOrderLineItems);
```

Reply:
오 이런 꿀팁이있네요!
객체 기본상태를 표현하니 가독성도 좋고 생성자도 감출 수 있네요!

기본 생성사는 private으로 만들고
`
    public static Order ofCooking(OrderTable orderTable, List<OrderLineItem> orderLineItems) {
        return new Order(orderTable, OrderStatus.COOKING, orderLineItems);
    }
`
를 생성해서 객체의 기본상태를 표현해주었습니다
감사합니다